frontend/packages/console-app/src/components/cloud-shell/CloudShellMastheadButton.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { Dispatch } from 'redux';
+import { connect } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { TerminalIcon } from '@patternfly/react-icons';
+import { isCloudShellExpanded } from '../../redux/reducers/cloud-shell-reducer';
+import { Button, ToolbarItem, Tooltip } from '@patternfly/react-core';
+import { connectToFlags, WithFlagsProps } from '@console/internal/reducers/features';
+import { confirmModal } from '@console/internal/components/modals/confirm-modal';
+import { FLAG_DEVWORKSPACE } from '../../consts';
+import { toggleCloudShellExpanded } from '../../redux/actions/cloud-shell-actions';
+import { useAccessReview } from '@console/internal/components/utils';
+import { WorkspaceModel } from '../../models';
+
+type DispatchProps = {
+  onClick: () => void;
+};
+
+type StateProps = {
+  open?: boolean;
+};
+
+type Props = WithFlagsProps & StateProps & DispatchProps;
+
+// TODO use proper namespace and resource name
+const namespace = 'che-workspace-controller';
+const name = 'cloudshell-userid';
+
+const ClouldShellMastheadButton: React.FC<Props> = ({ flags, onClick, open }) => {
+  const editAccess = useAccessReview({
+    group: WorkspaceModel.apiGroup,
+    resource: WorkspaceModel.plural,
+    verb: 'create',
+    name,
+    namespace,
+  });
+
+  const toggleTerminal = () => {
+    if (open) {
+      return confirmModal({
+        title: 'Close Terminal?',
+        message: (
+          <>
+            This will close the terminal session. Content in the terminal will not be resolved in
+            the next seesion.
+          </>
+        ),
+        btnText: 'Yes',
+        submitDanger: true,
+
+        cancelText: 'No',
+        executeFn: () => Promise.resolve(onClick()),
+      });
+    }
+    return onClick();
+  };
+
+  if (!editAccess || !flags[FLAG_DEVWORKSPACE]) {
+    return null;
+  }
+
+  return (
+    <Tooltip content={open ? 'Close Terminal' : 'Open Terminal'}>
```

Comment:
Tooltip content `Open command line terminal`. Check design doc.
Also, the tooltip should be on the button instead of `ToolbarItem`.